pyscriptjs/src/python/pyscript.py

```diff
@@ -124,6 +123,21 @@ def write(element_id, value, append=False, exec_id=0):
         )
 
 
+CURRENT_PY_SCRIPT_TAG = None
+
+
+def set_current_display_target(element):
+    global CURRENT_PY_SCRIPT_TAG
+    CURRENT_PY_SCRIPT_TAG = element
+
+
+def display(value, parent=None, append=True):
+    if parent is None:
+        global CURRENT_PY_SCRIPT_TAG
+        parent = CURRENT_PY_SCRIPT_TAG
+    Element(parent).write(value, append)
+
+
```

Comment:
I think it's ok to start from here but think there are ways we can improve and isolate `display` "current target" to the scope of execution so that we don't end up with race conditions of different scopes overwriting on each other. 

@philippjfr  suggestion of ContextVar in #769 is really interesting.

Reply:
Yeah!
I need @antocuni's input here on how to make it prettier.
I don't wanna use this kind of global 🙈

Reply:
But I intend to give it a try on the ContextVar stuff for sure =)
I'm updating the issue ^

Reply:
Nice! @antocuni , we have another opportunity do kindly disagree and trade beers! 😝

Reply:
> I think it's ok to start from here but think there are ways we can improve and isolate display "current target" to the scope of execution so that we don't end up with race conditions of different scopes overwriting on each other.

@fpliger 
sorry, reading it only now. I think that what you are proposing is impossible.
The semantics of Python scopes clashes with the desired semantics of `display()` in which the implicit target depends on the caller.
The only way to implement it (see below*) is to use a global variable. Or at least, that's the conclusion which I came up with after thinking *a lot* about the issue.
`ContextVar` helps with the `async` case but not in general, because it's essentially a "coroutine-aware global variable" -- so long as you stay in the same coroutine you have the very same problems as global variables.

I think that we will regret this choice: it's not really compatible with python semantics and will cause subtle bugs and problems in the future, but I'm not willing to fight for it :).

---
"the only way to implement it": well, the only reasonable way. The unreasonable way would be to modify the AST and/or the bytecode to detect the usage of `display()` and do some deep magic.
But if we go down this route, I'll resign immediately 😂

Reply:
@antocuni Apologies for butting in on this, and the following clarifying questions shouldn't delay this (quite good!) PR:

> The semantics of Python scopes clashes with the desired semantics of display() in which the implicit target depends on the caller... ContextVar helps with the async case but not in general, because it's essentially a "coroutine-aware global variable" -- so long as you stay in the same coroutine you have the very same problems as global variables.

I did want to clarify - what are the situations where you're in the same coroutine but you have different callers (different implicit targets)? Doesn't a single coroutine inPyScript have a single caller and therefore a well-defined implicit target that can be tracked via a ContextVar? 

What I think is probably best is for me to hush up for awhile, and have another spin at this once this PR is merged 😅

Reply:
> I did want to clarify - what are the situations where you're in the same coroutine but you have different callers (different implicit targets)?

that's my point, we don't know 😅.
Using global state to pass around function arguments is a bad API design: it makes functions not re-entrant and in general leads to subtle bugs and behavior. **Implicit** global state is even worse.
My point is that this approach is inherently fragile and it's "wrong by default": if we can think of all possible corner cases we can put workarounds for them, but as soon as we forget about one we will see bugs.

A random examples of things which might happen:
currently we do `set_display_target(X); ...; set_display_target(None)`, but if for *any reason* we end up calling this code recursively, the inner invocation would set the target to `None` and then the outer one would see it changing under its feet. AFAIK this is not something which can happen now but might happen in the future (maybe without even noticing). And also it would be easy to fix, you just need a stack of current targets instead of just a variable.
This might happen even sooner than we think: for example, what happens if you add a `<py-script>` tag to the DOM from python code? I don't know what is the desired semantics, but if by chance we decide that the python code needs to be executed immediately, then you are in the exact situation which I described above.

Another example is event handlers: when thinking about the issue I realized that event handlers were a problem, but god only knows how many other problematic situations there are that we couldn't think of.